src/gui/widgets/text_box.cpp

```diff
@@ -425,6 +426,7 @@ builder_text_box::builder_text_box(const config& cfg)
 	, max_input_length(cfg["max_input_length"])
 	, hint_text(cfg["hint_text"].t_str())
 	, hint_image(cfg["hint_image"])
+	, editable(cfg["editable"].to_bool())
```

Comment:
This is an API change, please don't include it in the 1.18 changes.

Also, the default doesn't come from the schema, it comes from the argument passed to `to_bool(bool def = false)`.

Reply:
We're in API and feature freeze, so in theory only changes that are required for bug fixing are allowed, but this whole PR is a case of "we really want the unit editor, so we'll waive that rule for stuff that's necessary to get it". Don't worry about it, I'd support this PR for the unit editor, and as you've seen @CelticMinstrel is more relaxed about changes that I am.

I'd probably be okay with adding it to `multiline_text` for 1.18 and moving it to the superclass in 1.19; would like to think about it a bit more before saying for sure. But at the same time that sounds like extra work and I'm not sure it's needed for 1.18.

Reply:
> I'd probably be okay with adding it to `multiline_text` for 1.18 and moving it to the superclass in 1.19; would like to think about it a bit more before saying for sure. But at the same time that sounds like extra work and I'm not sure it's needed for 1.18.

The `editable` key in `multiline_text` is need for the unit editor WML view. Also, if by superclass you mean `text_box_base`, then I've already added the `editable` boolean and corresponding getters and setters. (No WML)

I'll just remove the editable WML key from the text_box for now.